applications/MeshMovingApplication/custom_utilities/fixed_mesh_ale_utilities.cpp

```diff
@@ -346,54 +316,20 @@ namespace Kratos
         mpMeshMovingStrategy->SetEchoLevel(echo_level);
     }
 
-    const Vector FixedMeshALEUtilities::SetDistancesVector(ModelPart::ElementIterator ItElem) const
-    {
-        auto &r_geom = ItElem->GetGeometry();
-        Vector nodal_distances(r_geom.PointsNumber());
-
-        if (mLevelSetType == "continuous"){
-            // Continuous nodal distance function case
-            for (unsigned int i_node = 0; i_node < r_geom.PointsNumber(); ++i_node) {
-                nodal_distances[i_node] = r_geom[i_node].FastGetSolutionStepValue(DISTANCE);
-            }
-        } else if (mLevelSetType == "discontinuous") {
-            // Discontinuous elemental distance function case
-            nodal_distances = ItElem->GetValue(ELEMENTAL_DISTANCES);
-        } else {
-            KRATOS_ERROR << "Level set type must be either 'continuous' or 'discontinuous'. Got " << mLevelSetType;
-        }
-
-        return nodal_distances;
-    }
-
-    inline bool FixedMeshALEUtilities::IsSplit(const Vector &rDistances) const
-    {
-        unsigned int n_pos = 0, n_neg = 0;
-        for (double dist : rDistances) {
-            if(dist >= 0) {
-                ++n_pos;
-            } else {
-                ++n_neg;
-            }
-        }
-        if (n_pos > 0 && n_neg > 0) {
-            return true;
-        }
-        return false;
-    }
-
     void FixedMeshALEUtilities::InitializeVirtualMeshValues()
     {
-        // Copy the PRESSURE and VELOCITY values from the origin mesh to the virtual one
-        VariableUtils::Pointer p_var_utils = Kratos::make_shared<VariableUtils>();
-        for (unsigned int i_step = 0; i_step <  mpOriginModelPart->GetBufferSize(); ++i_step) {
-            p_var_utils->CopyModelPartNodalVar<Variable<double>>(PRESSURE, *mpOriginModelPart, mrVirtualModelPart, i_step);
-            p_var_utils->CopyModelPartNodalVar<Variable<array_1d<double,3>>>(VELOCITY, *mpOriginModelPart, mrVirtualModelPart, i_step);
+        // Initialize the MESH_DISPLACEMENT and MESH_VELOCITY values
+        // Note that both positions of the buffer are initialized to zero. This is important
+        // in case the CloneTimeStep() is done in the virtual model part, since the method
+        // assumes that the mesh is in the origin position when computing the MESH_VELOCITY.
+        #pragma omp parallel for
+        for (int i_node = 0; i_node < static_cast<int>(mrVirtualModelPart.NumberOfNodes()); ++i_node) {
+            auto it_node = mrVirtualModelPart.NodesBegin() + i_node;
+            it_node->FastGetSolutionStepValue(MESH_VELOCITY, 0) = ZeroVector(3);
```

Comment:
why not in all the buffer?

Reply:
Because a BDF1 scheme is always used to compute the `MESH_VELOCITY`. Therefore only the positions 0 and 1 of the buffer are required.

(maybe in the future we can make this optional)